vividus-plugin-web-app/src/main/java/org/vividus/selenium/screenshot/AshotFactory.java

```diff
@@ -97,30 +100,27 @@ private ShootingStrategy decorateWithViewportPasting(ShootingStrategy toDecorate
                        .withScrollTimeout(((Long) screenshotConfiguration.getScrollTimeout().toMillis()).intValue());
     }
 
-    private AShot createAShot(ShootingStrategy baseShootingStrategy, boolean viewportScreenshot)
-    {
-        return createAShot(baseShootingStrategy, screenshotShootingStrategy, viewportScreenshot);
-    }
-
-    private AShot createAShot(ShootingStrategy baseShootingStrategy,
-            ScreenshotShootingStrategy screenshotShootingStrategy, boolean viewportScreenshot)
+    private AShot createAShot(String screenshotShootingStrategyName)
     {
+        ScreenshotShootingStrategy configured = getStrategyBy(screenshotShootingStrategyName);
         String deviceName = webDriverFactory.getCapability(MobileCapabilityType.DEVICE_NAME, false);
         boolean landscapeOrientation = webDriverManager.isOrientation(ScreenOrientation.LANDSCAPE);
-        ShootingStrategy shootingStrategy = screenshotShootingStrategy.getDecoratedShootingStrategy(
-                baseShootingStrategy, viewportScreenshot, landscapeOrientation, deviceName);
-        return new AShot().shootingStrategy(shootingStrategy)
-                .coordsProvider(screenshotShootingStrategy == ScreenshotShootingStrategy.SIMPLE
+        ShootingStrategy baseShootingStrategy = getBaseShootingStrategy();
+        ShootingStrategy shootingStrategy = configured.getDecoratedShootingStrategy(
+                baseShootingStrategy, landscapeOrientation, deviceName);
+        return new ScrollbarHidingAshot(Optional.empty(), scrollbarHandler).shootingStrategy(shootingStrategy)
+                .coordsProvider(configured instanceof SimpleScreenshotShootingStrategy
                         ? CeilingJsCoordsProvider.getSimple(javascriptActions)
                         : CeilingJsCoordsProvider.getScrollAdjusted(javascriptActions));
     }
 
-    private boolean anyNotZero(int first, int second)
+    @Override
+    protected float getDpr()
     {
-        return first > 0 || second > 0;
+        return (float) javascriptActions.getDevicePixelRatio();
```

Comment:
why it should be `float`?

Reply:
Brand | Device | Resolution | Pixel Ratio
-- | -- | -- | --
Amazon | Kindle Fire | 1024x600 | 1
Amazon | Kindle Fire HD 7" | 1280x800 | 1.5
Amazon | Kindle Fire HD 8.9" | 1920x1200 | 1.5
Apple | iPad 1 | 1024x768 | 1
Apple | iPad 2 | 1024x768 | 1
Apple | iPad 3 | 2048x1536 | 2
Apple | iPad 4 | 2048x1536 | 2
Apple | iPad Mini | 1024x768 | 1
Apple | iPhone 3GS | 320x480 | 1
Apple | iPhone 4 | 640x960 | 2
Apple | iPhone 5 | 640x1136 | 2
BlackBerry | PlayBook | 1024x600 | 1
BlackBerry | Z10 | 768x1280 | 2
BlackBerry | Z30 | 720x1280 | 2
Google | Nexus 10 | 2560x1600 | 2
Google | Nexus 4 | 768x1280 | 2
Google | Nexus 5 | 1080x1920 | 3
Google | Nexus 7 | 1280x800 | 1.325
Google | Nexus 7 (2013) | 1920x1200 | 2
Google | Nexus S | 480x800 | 1.5
HTC | Desire | 480x800 | 1.5
HTC | Desire HD | 480x800 | 1.5
HTC | Evo | 480x800 | 1.5
HTC | Evo 3D | 540x960 | 1.5
HTC | EVO LTE | 720x1280 | 2
HTC | One X | 720x1280 | 2
HTC | Sensation | 540x960 | 1.5
HTC | Touch HD | 480x800 | 1.5
LG | Optimus 2X | 480x800 | 1.5
LG | Optimus 3D | 480x800 | 1.5
LG | Optimus 4X HD | 720x1280 | 1.7
LG | Optimus Black | 480x800 | 1.5
LG | Optimus G | 768x1280 | 2
LG | Optimus LTE | 720x1280 | 1.7
LG | Optimus One | 320x480 | 1.5
Motorola | Atrix 2 | 540x960 | 1
Motorola | Atrix 4G | 540x960 | 1
Motorola | Defy | 480x854 | 1.5
Motorola | Droid | 480x854 | 1.5
Motorola | Droid 3 | 540x960 | 1
Motorola | Droid 4 | 540x960 | 1
Motorola | Droid Razr | 540x960 | 1
Motorola | Droid Razr HD | 720x1280 | 1
Motorola | Droid X | 480x854 | 1.5
Motorola | Milestone | 480x854 | 1.5
Motorola | Xoom | 1280x800 | 1
Motorola | Xyboard | 1280x800 | 1
Nokia | C5 | 360x640 | 1
Nokia | C6 | 360x640 | 1
Nokia | C7 | 360x640 | 1
Nokia | Lumia 7X0 | 480x800 | 1.5
Nokia | Lumia 8XX | 480x800 | 1.5
Nokia | Lumia 900 | 480x800 | 1.5
Nokia | Lumia 920 | 768x1280 | 2.4
Nokia | Lumia 925 | 768x1280 | 2.4
Nokia | Lumia 928 | 768x1280 | 2.4
Nokia | Lumia 1020 | 768x1280 | 2.4
Nokia | N8 | 360x640 | 1
Nokia | N800 | 480x800 | 1.5
Nokia | N810 | 480x800 | 1.5
Nokia | N900 | 480x800 | 1.5
Nokia | N97 | 360x640 | 1
Nokia | X7 | 360x640 | 1
Samsung | Galaxy Nexus | 720x1280 | 2
Samsung | Galaxy Note | 800x1280 | 2
Samsung | Galaxy Note 2 | 720x1280 | 2
Samsung | Galaxy Note 3 | 1080x1920 | 2
Samsung | Galaxy S | 480x800 | 1.5
Samsung | Galaxy S 2 | 480x800 | 1.5
Samsung | Galaxy S 3 | 720x1280 | 2
Samsung | Galaxy S 4 | 1080x1920 | 3
Samsung | Galaxy Tab | 1024x600 | 1
Samsung | Galaxy Tab 10.1 | 1280x800 | 1
Samsung | Galaxy Tab 7.7 | 1280x800 | 1
Samsung | Galaxy Tab 8.9 | 1280x800 | 1
Samsung | Galaxy W | 480x800 | 1.5
Sony | Xperia Ion | 720x1280 | 2
Sony | Xperia S | 720x1280 | 2
Sony | Xperia Sola | 480x854 | 1
Sony | Xperia U | 480x854 | 1
Sony | Xperia Z | 1080x1920 | 3
Sony | Xperia Z1 | 1080x1920 | 3

Reply:
it's `double` originally, but casted to `float`

Reply:
https://github.com/pazone/ashot/blob/41a81919b40cc745d0652eae487e1f2ff54c5c41/src/main/java/pazone/ashot/ScalingDecorator.java#L47